### ISSUE TITLE
[Request] Computer Science topics outside of programming

### ISSUE THREAD

**Issue by gizmo385** [adambard/learnxinyminutes-docs#526]
I think the addition of pages on things related to programming and computer science, but not purely programming would be helpful. For example, a page covering sorting algorithms and their differences or a page on basic algorithm analysis.

Thoughts?

**Comment by Jakehp** [comment]
@levibostian @adambard I started this today. I am thinking we should create a new section for "Algorithms & Data Structures". What do you think?

**Comment by levibostian** [comment]
@Jakehp you have my vote!

**Comment by Jakehp** [comment]
https://github.com/adambard/learnxinyminutes-docs/pull/950

**Comment by gizmo385** [comment]
Definite (albeit late) +1 on the contribution by @Jakehp. Is there any way we'd be able to use MathJax or something similar to parse LaTeX? Once the explanations start delving deeper into some of the more complicated math related to algorithms and data structures, certain things like sums are going to become commonplace and we'll need a way to effectively express those to the users. Thoughts?

**Comment by Jakehp** [comment]
@gizmo385 That's a really good idea; although I'll have to learn LaTeX. @adambard is this possible?

**Comment by levibostian** [comment]
I've experienced MathJax.js in the past and I really liked it. Just a simple [js file](https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML) you stick in the head and anywhere in your HTML you enclose Latex. 

Here is a fully working example:  
```
<!DOCTYPE html>
<html>
<head>
<title>MathJax TeX Test Page</title>
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});
</script>
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
</head>
<body>
When $a \ne 0$, there are two solutions to \(ax^2 + bx + c = 0\) and they are
$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}.$$
</body>
</html>
```

So really, we could do that pretty easy.

**Comment by geoffliu** [comment]
How would that play with other pages? Would that screw up, for example, the Perl page?

Can we configure that somehow to use `<latex>` tags instead of `$ $`?

**Comment by geoffliu** [comment]
Well, using `$ $` is a bit dangerous, especially for the Perl page, no?

I would much prefer to have `<latex>` tags for these, and looks like the config like lets you specify any delimiter?

**Comment by iirelu** [comment]
@levibostian Could I recommend [KaTeX](https://khan.github.io/KaTeX/) instead of MathJax? It's leagues faster and can also be rendered server-side.

**Comment by gizmo385** [comment]
I've heard similar things about KaTeX @iirelu. Would we be able to take advantage of the server side rendering so that we'd only ever have to generate the corresponding render for any given LaTeX formula once?

As for the tags @geoffliu, it looks as though you can customize the inline math tags. I haven't worked with MathJax (or KaTeX) before, but I would assume that we would be able to come up with a suitable tag that we'd be able to use.

**Comment by Oire** [comment]
My two cents :). It would be really good if any maths notation you select would be accessible to everyone, and namely to the blind and visually impaired people. @iirelu, any thoughts, please? :)

**Comment by iirelu** [comment]
@Oire What do you mean by accessibility? It doesn't look to me like MathJax has any accessibility improvements over KaTeX.

**Comment by Oire** [comment]
@iirelu, what would you get after rendering KaTeX? As for MathJax, latest versions of screen reading solutions (namely, JAWS 16) seem to support (at least basic) MathML rendered by MathJax. Am I totally mistaken and out of subject here?)  
P.S. Sorry if it is not your strong point, but I saw a neat little thing making Braille from images in your profile, so I thought you might be more informed than most people :).

**Comment by iirelu** [comment]
@Oire Server-side rendering just renders HTML, so the browser displays the exact same thing as if KaTeX was running locally except it works even with javascript disabled.

**Comment by Jakehp** [comment]
Guess I'll wait on writing next articles until we settle this?

**Comment by iirelu** [comment]
@Jakehp No, go ahead, things can be fixed later if need be.

**Comment by adambard** [comment]
Totally, go nuts. I'm probably most likely to find a way to render latex when there's a pressing need to render latex :)

**Comment by NickPapanastasiou** [comment]
Data structures and algorithms are a sweet addition, but how about slightly more niche topics? For example, if someone were to present as a learnxinyminutes page an overview of compilers, operating system principles, game engines, or software  design patterns?

**Comment by quantomistro** [comment]
I really love the addition of these new topics! But, I think the category name of "Algorithms & Data Structures" doesn't fit these topics very much. Maybe something like "Theory"? The topics focus more on CS theory, rather than just algorithms & data structures.
